src/components/AddModalForm.js

```diff
@@ -0,0 +1,211 @@
+import React, { useState } from 'react';
+import { Form, Modal, InputGroup, Button } from 'react-bootstrap';
+
+export default function AddModalForm(props) {
+  const { modal, setModal, user, token } = props;
+  const [formData, setFormData] = useState({
+    name: '',
+    ingredients: [''],
+    method: [''],
+    imageFile: '',
+    cookTime: 0,
+  });
+  const [validated, setValidated] = useState(false);
+
+  const onSubmit = (event) => {
+    const form = document.getElementById('add-recipe-form');
+    if (form.checkValidity()) {
+      fetch('/recipes', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name: formData.name,
+          ingredients: formData.ingredients,
+          method: formData.method,
+          time: formData.cookTime,
+          createdBy: user.name,
+        }),
+      }).catch((err) => console.log('Error in adding recipe: ' + err));
+      setModal(false);
+    } else {
+      event.preventDefault();
+      event.stopPropagation();
+    }
+    setValidated(true);
+  };
+
+  return (
```

Comment:
as an aside, i feel like add recipe should be the same as edit(null). From a client perspective, it's all the same fields except add doesn't populate anything. From a server perspective, if something comes in without an id, then it's add.

Reply:
i believe I address this in a later PR